packages/devtools-reps/src/object-inspector/components/ObjectInspector.js

```diff
@@ -93,21 +97,34 @@ class ObjectInspector extends Component<Props> {
       if (this.props.rootsChanged) {
         this.props.rootsChanged();
       }
+      return;
+    }
+
+    // if there are new evaluations, we want to remove the existing cached
+    // nodes from the cache.
+    if (nextProps.evaluations.size > this.props.evaluations.size) {
```

Comment:
could we do an identity check?

Reply:
maybe move this up to a small utility `removeOldNodes`

Reply:
> could we do an identity check?

sure

> maybe move this up to a small utility

okay